modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -93,12 +93,12 @@ interface WalkResult {
 }
 
 // See https://html.spec.whatwg.org/multipage/parsing.html#parsing-html-fragments
-// for anything special excluding elements that use the RCDATA state
-const specialElements = new Set('style,xmp,iframe,noembed,noframes,script,noscript,plaintext'.split(','));
+const specialElements = new Set('title,textarea,style,xmp,iframe,noembed,noframes,script,noscript,plaintext'.split(','));
```

Comment:
Should this be moved to a central schema at some point? I really want to avoid us having hard coded lists of elements all over the place if we can.

Reply:
No, as it's not related to the schema in any form... it's related to the HTML parsing spec and how browsers parse, so it's only applicable in this module.

Reply:
I guess we need to discuss how we are to handle the html schema in general in the future. Since in the new alt model it's a dynamic instance being passed around and that is used for editing behavior rules and model normalization:
https://github.com/tinymce/rockfest/blob/master/modules/bs-slate/src/main/re/schema/Slate__Core__Schema.resi

For this case it's probably fine to have a hard coded list of things since they are parser specific and likely not going to change that often. But I want to avoid having lists of elements all over since browsers add things to the spec all the time. However I am guilty of adding some hard coded lists my self mostly used by the content editable support:
https://github.com/tinymce/tinymce/blob/develop/modules/tinymce/src/core/main/ts/dom/ElementType.ts
However since we are switching to a model that has a dynamic schema again and the contenteditable support is going to be rewritten from scratch then that thing will not be used anymore.

The purpose of the Schema when it was designed was to have one spot for all parser rules so it used to be the central place of the HTML spec rules configurable and static ones but also for editing behavior one rule set for everything. So to me it is very much a part of the HTML parser that is why it is passed into the DomParser. This special element could have been a predicate that takes a string on that Schema instance. However not sure this needs to be public api it's unlikely that some plugin needs this information. So I guess it makes sense to hard code the list here then.

However it might be useful to have this for the normalization of the model not just for parsing. Since these elements should never be allowed to have a child elements they should all be converted into string or dropped if they are appended and since the valid children is a dynamic thing so someone could configure a iframe to allow child elements. Having a fixed set there to block that from happening might be useful. But I getting ahead of my self.

Reply:
You're getting confused here sorry Spocke, as we still have the schema and it's used for validation as it previously was. This however relates to how the native parser parses and as such **cannot** be something that can be configured by a custom schema.

Reply:
To clarify further `Slate__Core__Schema.resi` is not related to this piece of code and it instead relates to `Schema.ts`. The custom schema is then used by both DOMPurify to remove elements that aren't allowed and then further once it's been transformed into an AST to clean up invalid children as defined by the custom schema.

Reply:
No I'm not confused at all. But I may not be good at explaining what I mean. Parts of the schema is dynamic parts of it is static now it's mostly dynamic it doesn't have to be. I say all HTML rules dynamic or static should be in one central place or perhaps two central places not N places. Since the rules here for the parser might need to be applied to normalization or other things. It's information that might be useful outside the parser it's information about how HTML elements behave in general. I guess void elements should also be a fixed set that the user can't change but that is also useful outside parsing like making atomic selections of them etc or normalizing them if they have elements inside.

Reply:
Sorry for the late reply here, but okay I get where you're coming from now so thanks for persevering. I still don't agree however for this specific case (though I do for most other things), as while the two are similar they are different as the custom schema is not a 1:1 mapping of how the native parser works and the rest of TinyMCE operates or uses the schema. That said, I don't think it's a huge issue either way so for now I've made it reuse the `schema.getSpecialElements()` API, added a TODO to freeze it when we can and made it so it's not deprecated since we're going to need to maintain it if you want this.

Reply:
Thanks yes it should be frozen and so should the default set of void elements be for example img should always be a void but you should be able to extend it with custom elements that become atomic units. If we have one schema passed around for everything including parsing it one spot to change, one rule set not multiple ones. It's more how we want the schema to work in the future than how it works today where nearly all of it is dynamic and reconfigurable. I regret adding options for configuring some of these things that should be hard coded that is why they where never documented.